java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/WorkspaceServiceImpl.java

```diff
@@ -1292,6 +1292,7 @@ public void didChangeConfiguration(DidChangeConfigurationParams params) {
         String fullConfigPrefix = client.getNbCodeCapabilities().getConfigurationPrefix();
         String configPrefix = fullConfigPrefix.substring(0, fullConfigPrefix.length() - 1);
         server.openedProjects().thenAccept(projects -> {
+            ((TextDocumentServiceImpl)server.getTextDocumentService()).updateJavaHintPreferences(((JsonObject) params.getSettings()).getAsJsonObject("netbeans").getAsJsonObject("hints"));
```

Comment:
Please use `configPrefix` instead of `"netbeans"`. Different extensions have different configuration prefixes.

And, maybe, also use `NETBEANS_JAVA_HINTS` instead of `"hints"`?